src/common/helpers/search.js

```diff
@@ -374,8 +384,22 @@ export async function generateIndex(orm) {
 	const entityLists = await Promise.all(behaviorPromise);
 
 	const listIndexes = [];
-	for (const entityList of entityLists) {
+	for (const [index, entityList] of entityLists.entries()) {
 		const listArray = entityList.toJSON();
+		if (index === (entityLists.length - 1)) {
```

Comment:
It would be better and more future-proof to check by entity type name rather than position in an array.
If the order changes in the future this will break.